IFC_Adapter/Methods/Convert/Environment/FromIFC/Space.cs

```diff
@@ -43,10 +44,19 @@ public static Space SpaceFromIfc(this IIfcSpace element, IfcSettings settings)
                 return null;
             }
 
+            IfcObject ifcObject = element as IfcObject;
+            if (ifcObject == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("The IFC element could not be read due to an internal error.");
```

Comment:
I'd change the error text to something like
```suggestion
                BH.Engine.Reflection.Compute.RecordError($"The provided IFC element of type {element.GetType().Name} could not be converted to {nameof(IfcObject)}.");
```

Reply:
Fair point. Actually, casting to `IfcObject` is likely to be needed in every method, so makes sense to centralise it along with the error in a separate method.